src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala

```diff
@@ -239,6 +245,43 @@ private[spark] class KubernetesSuite extends FunSuite with BeforeAndAfterAll wit
       appLocator)
   }
 
+  private def runThriftServerAndVerifyQuery(
+                                             driverPodChecker: Pod => Unit = doBasicDriverPodCheck,
+                                             appArgs: Array[String] = Array.empty[String],
+                                              appLocator: String = appLocator): Unit = {
+    val appArguments = SparkAppArguments(
+      mainAppResource = "",
+      mainClass = "org.apache.spark.sql.hive.thriftserver.HiveThriftServer2",
+      appArgs = appArgs)
+    SparkAppLauncher.launch(appArguments, sparkAppConf, TIMEOUT.value.toSeconds.toInt, sparkHomeDir)
+    val driverPod = kubernetesTestComponents.kubernetesClient
+      .pods
+      .withLabel("spark-app-locator", appLocator)
+      .withLabel("spark-role", "driver")
+      .list()
+      .getItems
+      .get(0)
+    driverPodChecker(driverPod)
+    val driverPodResource = kubernetesTestComponents.kubernetesClient
+      .pods
+      .withName(driverPod.getMetadata.getName)
+    
+    Eventually.eventually(TIMEOUT, INTERVAL) {
+      val localPort = driverPodResource.portForward(10000).getLocalPort
+      val jdbcUri = s"jdbc:hive2://localhost:$localPort/"
+      val connection = DriverManager.getConnection(jdbcUri, "user", "pass")
+      val statement = connection.createStatement()
+      try {
+        val resultSet = statement.executeQuery("select 42")
+        resultSet.next()
+        assert(resultSet.getInt(1) == 42)
+      } finally {
+        statement.close()
+        connection.close()
```

Comment:
`connection` should be closed regardless of if `statement.close` throws an exception.

Reply:
Fair enough. I got this from https://github.com/apache/spark/blob/master/sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/JdbcConnectionUriSuite.scala#L66, worth fixing?